app/models/amusement_park.rb

```diff
@@ -1,3 +1,8 @@
 class AmusementPark < ApplicationRecord
   has_many :rides
+  has_many :mechanics, through: :rides
+
+  def unique_list_of_mechanics
+    self.mechanics.distinct
+  end
```

Comment:
Whoot! Really good job adding another association to transfer the responsibility of doing the joins over to Rails.